WatchFaceKotlin/wear/src/main/java/com/example/android/wearable/watchfacekotlin/watchface/AnalogComplicationWatchFaceService.kt

```diff
@@ -0,0 +1,834 @@
+         */
+        @InternalCoroutinesApi
+        private fun startSecondHandAnimation() {
+            if (!secondHandAnimationActive()) {
+                secondHandAnimationJob = watchfaceCoroutineScope.launch(Dispatchers.Main) {
+                    secondHandAnimationFlow().collect {
+                        // Redraws second hand one second apart (triggered by flow)
+                        invalidate()
+                    }
+                }
+            }
+        }
+
+        /**
+         * Stops the coroutine/flow "timer" when the watch face isn't visible (or is in ambient
+         * mode).
+         */
+        private fun stopSecondHandAnimation() {
+            if (secondHandAnimationJob?.isActive == true) {
+                secondHandAnimationJob?.cancel()
+            }
+        }
+
+        /**
+         * Second hand should only be visible in active mode.
+         */
+        private fun secondHandVisible(): Boolean {
+            return isVisible && !ambient
+        }
+
+        private fun secondHandAnimationActive(): Boolean {
+            return secondHandAnimationJob?.isActive ?: false
+        }
+
+        /**
+         * Creates a flow that triggers one second apart. It's used to animate the second hand when
+         * the watch face is in active mode.
+         */
+        private fun secondHandAnimationFlow(): Flow<Unit> = flow {
+            while (secondHandVisible()) {
+                val timeMs = System.currentTimeMillis()
+                Log.d(TAG, "timeMs: $timeMs")
+                val delayMs = INTERACTIVE_UPDATE_RATE_MS - (timeMs % INTERACTIVE_UPDATE_RATE_MS)
+                Log.d(TAG, "delayMs: $delayMs")
+                delay(delayMs)
+                // Emit next value
+                emit(Unit)
+            }
+        }
+    }
+
+    companion object {
+        private const val TAG = "AnalogWatchFace"
+
+        /*
+         * Update rate in milliseconds for interactive mode. We update once a second to advance the
+         * second hand.
+         */
+        private val INTERACTIVE_UPDATE_RATE_MS = TimeUnit.SECONDS.toMillis(1)
+
+        private const val HOUR_STROKE_WIDTH = 5f
+        private const val MINUTE_STROKE_WIDTH = 3f
+        private const val SECOND_TICK_STROKE_WIDTH = 2f
+        private const val CENTER_GAP_AND_CIRCLE_RADIUS = 4f
+        private const val SHADOW_RADIUS = 6
+
+
+        // Unique IDs for each complication. The settings activity that supports allowing users
+        // to select their complication data provider requires numbers to be >= 0.
+        private const val BACKGROUND_COMPLICATION_ID = 0
+        private const val LEFT_COMPLICATION_ID = 100
+        private const val RIGHT_COMPLICATION_ID = 101
+
+        // Used by {@link AnalogComplicationConfigRecyclerViewAdapter} to retrieve all complication
+        // ids. Background, Left and right complication IDs as array for Complication API.
+        val complicationIds = intArrayOf(
```

Comment:
I'm struggling a bit to figure out which complication goes with which id, with which set of supported Types. It looks like a Sealed class might be able to help here. 
```
sealed class ComplicationConfig(val id, val supportedTypes) {
  class Left() : ComplicationConfig(LEFT_COMPLICATION_ID, intArrayOf(...)
  ...
```